extension/source/QuillPage/components/ViewSecretPhrasePanel.tsx

```diff
@@ -42,48 +43,48 @@ const ViewSecretPhrasePanel: React.FunctionComponent = () => {
           restoring your wallets should you lose access.
         </p>
       </div>
-      <div className="secret-phrase-box">
-        {Range(4).map((i) => (
-          <QuickRow key={`row${i}`}>
-            {Range(3).map((j) => (
-              <QuickColumn key={`column${j}`}>
-                {3 * i + j + 1}. {secretWords[3 * i + j]}
-              </QuickColumn>
-            ))}
+      {expanded && (
+        <div className="secret-phrase-box">
+          {Range(4).map((i) => (
+            <QuickRow key={`row${i}`}>
+              {Range(3).map((j) => (
+                <QuickColumn key={`column${j}`}>
+                  {3 * i + j + 1}. {secretWords[3 * i + j]}
+                </QuickColumn>
+              ))}
+            </QuickRow>
+          ))}
+        </div>
+      )}
+      {!expanded && (
+        <div className="show-box">
+          <div style={{ display: 'inline-block' }}>
+            <Button
+              onPress={() => setExpanded(true)}
+              highlight={false}
+              icon={<Eye className="icon-md" />} // TODO: Is icon-md needed?
+            >
+              Show secret phrase
+            </Button>
+          </div>
+        </div>
+      )}
+      {expanded && (
+        <div className="hide-box">
+          <QuickRow>
+            <Button onPress={() => setExpanded(false)} highlight={false}>
+              Hide secret phrase
+            </Button>
+            <Button
+              onPress={() => {}}
+              highlight={true}
+              icon={<ArrowRight className="icon-md" />}
+            >
+              Review secret phrase
+            </Button>
```

Comment:
Hey this block is unnecessarily too long
This suggestion handles the same thing in much less lines of code, happy to discuss.
https://github.com/jzaki/bls-wallet/pull/93#discussion_r783929591

Reply:
the hyperlink doesn't seem to work. Here's the code snippet -

```tsx
    <QuickRow>
        <div style={{ display: 'inline-block' }}>
          <Button
            onPress={() => setExpanded(!expanded)}
            highlight={false}
            icon={{
              src: browser.runtime.getURL('assets/eye.svg'),
              px: 19,
            }}
          >
            {expanded ? "Hide secret phrase" : "Show secret phrase"}
          </Button>
        </div>
        {expanded && <Button
            onPress={() => {}}
            highlight={true}
            icon={{
              src: browser.runtime.getURL('assets/arrow-small.svg'),
              px: 19,
            }}
          >
            Review secret phrase
          </Button>}
     </QuickRow>
```

Reply:
Not sure what you mean by the hyperlink? If you mean the `Review secret phrase` button, that's added in the next PR.

Reply:
> Hey this block is unnecessarily too long

I've implemented this because I don't mind the combined button approach but I don't think there was anything wrong there. Less code doesn't mean better code. Reviews on the other hand, benefit greatly if they can be made shorter.